File: Data Structures & Algorithms/three-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    threeSum(nums) {
        let res = [];
        nums.sort((a,b) => a - b)
        for(let i = 0; i < nums.length; i++){
            if(i > 0 && nums[i] == nums[i - 1]){
            continue;}
            let r = nums.length - 1;
            let l = i + 1;
            while(l < r){
                let threesum = nums[i] + nums[l] + nums[r];
                if(threesum > 0)
                {
                    r -= 1
                }
                else if(threesum < 0){
                    l += 1
                }
                else{
                    res.push([nums[i],nums[r],nums[l]])
                    l += 1;
                    while(nums[l] == nums[l-1]&& l < r){
                        l += 1;
                    }
                }
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[0])
        last = intervals[0]
        res = 0
        for i in intervals[1:]:
            if i[0] < last[1]:
                last[1] = min(i[1],last[1])
                print("we are in")
                print(last,i)
                res += 1
            else:
                last = i
        return(res)
        